xml/hwbp_registers.xml

```diff
@@ -155,21 +155,27 @@
     </register>
 
     <register name="Hypervisor Context" short="hcontext" address="0x6a8">
-        This optional register is only writable in HS mode, M mode and Debug Mode.
+        This optional register is only accessible in HS mode, M mode and Debug Mode.
+
+        If the H extension is not implemented then this register is not
+        implemented, though the underlying state may be accessible via
+        the optional \RcsrMcontext alias.
 
         <field name="hcontext" bits="XLEN-1:0" access="WARL" reset="0">
             Hypervisor mode software can write a context number to this register,
             which can be used to set triggers that only fire in that specific
             context.
 
             An implementation may tie any number of upper bits in this field to
-            0. It's recommended to implement no more than 6 bits on RV32, and
-            13 on RV64.
+            0. If the H extension is not supported, it's recommended to implement
+            no more than 6 bits on RV32, and 13 on RV64.  If the H extension is
+            supported, it's recommended to implement no more than 7 bits on RV32
+            and 14 on RV64.
         </field>
     </register>
 
     <register name="Supervisor Context" short="scontext" address="0x5a8">
-        This optional register is only writable in S mode, M mode and Debug Mode.
+        This optional register is only accessible in S mode, M mode and Debug Mode.
```

Comment:
I think this register is also accessible in HS mode and VS mode, right?

Reply:
First off, at a minimum this should say "S/HS mode", i.e. they are different names - based on the context of whether the H-extension is implemented and enabled - for the same privilege level.  If the H extension is enabled, then S-mode is now called HS-mode and certain additional instructions and CSRs become available (on top of all the standard S-mode arch functionality).

Regarding VS mode, the H-extension generally (i.e. in most but not all cases) takes the approach that there  are s* and vs* versions of what originally were just s* CSRs.  They are physically distinct registers.  In HS-mode both are accessible and the s* registers are active (in having whatever effects they are defined to have).  In VS-mode vs* CSRs are inaccessible directly, but they become "active" and are accessed as s* CSRs (in place of the normal s* CSRs which are inactive in VS-mode).  In other words, the vs* CSRs masquerade as the s* CSRs in VS-mode.

In this case we either add a vscontext CSR that behaves as just described (but is that cost worth it?), or we stick with just the current scontext CSR and make it accessible in both S/HS and VS modes.  I believe in past email discussion we decided to take the latter approach.  The hypervisor will context switch scontext when switching between VMs (just as it would have done with vscontext).  If scontext is not being used within HS and U modes, then that's it.  Otherwise the hypervisor also needs to context switch scontext between HS/U mode usage and VS/VU mode usage.

So, in short, scontext should be accessible in S/HS mode, VS mode, M mode, and Debug mode.

Reply:
I added that it's accessible in VS mode.  I also changed the mode filter bits in various CSRs to refer to S/HS mode instead of just S mode (since HS and S share the same filter bit depending on whether hypervisor is supported).